lib/improver/spotdata/tests/test_neighbour_finding.py

```diff
@@ -0,0 +1,679 @@
+        """Create a cube containing a regular lat-lon grid."""
+        data = np.zeros((20, 20))
+        latitudes = np.linspace(-90, 90, 20)
+        longitudes = np.linspace(-180, 180, 20)
+        latitude = DimCoord(latitudes, standard_name='latitude',
+                            units='degrees')
+        longitude = DimCoord(longitudes, standard_name='longitude',
+                             units='degrees')
+
+        cube = Cube(data,
+                    long_name="test_data",
+                    dim_coords_and_dims=[(latitude, 0), (longitude, 1)],
+                    units="1")
+
+        orography = cube.copy()
+        orography.rename('surface_altitude')
+        land = cube.copy()
+        land.rename('land_binary_mask')
+        land.data = land.data + 1
+
+        ancillary_data = {}
+        ancillary_data.update({'orography': orography})
+        ancillary_data.update({'land': land})
+
+        sites = OrderedDict()
+        sites.update({'100': {'latitude': 50,
+                              'longitude': 0,
+                              'altitude': 10,
+                              'gmtoffset': 0
+                              }
+                      })
+
+        neighbour_list = np.empty(1, dtype=[('i', 'i8'),
+                                            ('j', 'i8'),
+                                            ('dz', 'f8'),
+                                            ('edge', 'bool_')])
+
+        self.cube = cube
+        self.ancillary_data = ancillary_data
+        self.sites = sites
+        self.neighbour_list = neighbour_list
+
+    def return_types(self, method):
+        """Test that the plugin returns a numpy array."""
+        plugin = PointSelection(method)
+        result = plugin.process(self.cube, self.sites,
+                                ancillary_data=self.ancillary_data)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertEqual(result.dtype, self.neighbour_list.dtype)
+
+    def correct_neighbour(self, method, i_expected, j_expected, dz_expected):
+        """Test that the plugin returns the expected neighbour"""
+        plugin = PointSelection(method)
+        result = plugin.process(self.cube, self.sites,
+                                ancillary_data=self.ancillary_data)
+        self.assertEqual(result['i'], i_expected)
+        self.assertEqual(result['j'], j_expected)
+        self.assertEqual(result['dz'], dz_expected)
+
+    def without_ancillary_data(self, method):
+        """Test plugins behaviour with no ancillary data provided"""
+        plugin = PointSelection(method)
+        if method == 'fast_nearest_neighbour':
+            result = plugin.process(self.cube, self.sites)
+            self.assertIsInstance(result, np.ndarray)
+        else:
+            msg = 'Ancillary data'
+            with self.assertRaisesRegexp(Exception, msg):
+                result = plugin.process(self.cube, self.sites)
+
+
+class fast_nearest_neighbour(TestNeighbourFinding):
```

Comment:
A bactrian following a dromedary (or the other way around). Are you requesting the class names become CamelCase as per the style guide? The reason I haven't is so they directly correspond to the methods that they are testing. I thought it was clearer, even if away from the style guide. Happy to change if you think I should.